mllib/src/test/scala/org/apache/spark/ml/source/libsvm/LibSVMRelationSuite.scala

```diff
@@ -184,4 +184,22 @@ class LibSVMRelationSuite extends SparkFunSuite with MLlibTestSparkContext {
       spark.sql("DROP TABLE IF EXISTS libsvmTable")
     }
   }
+
+  test("SPARK-32815: Test LibSVM data source on file paths with glob metacharacters") {
+    val basePath = Utils.createDirectory(tempDir.getCanonicalPath, "globbing")
```

Comment:
Here is the diff with the patch for 3.0 because withTempDir is not available in 2.4 here:
https://github.com/apache/spark/pull/29675/files#diff-ed6960b663c36ea39eee8c20aae09fb3R196